lib/SPIRV/libSPIRV/SPIRVModule.cpp

```diff
@@ -138,8 +140,6 @@ class SPIRVModuleImpl : public SPIRVModule {
   void setAlignment(SPIRVValue *, SPIRVWord);
   void setMemoryModel(SPIRVMemoryModelKind MM) {
     MemoryModel = MM;
-    if (MemoryModel == spv::MemoryModelOpenCL)
-      addCapability(CapabilityKernel);

```

Comment:
It would be cleaner to keep setMemoryModel() unchanged whereas call it in the SPIRVModuleImpl constructor.